diff --git a/include/linux-private/linux/socket.h b/include/linux-private/linux/socket.h
index 268b948..90a26a3 100644
--- a/include/linux-private/linux/socket.h
+++ b/include/linux-private/linux/socket.h
@@ -2,21 +2,7 @@
 #ifndef _LINUX_SOCKET_H
 #define _LINUX_SOCKET_H
 
-/*
- * Desired design of maximum size and alignment (see RFC2553)
- */
-#define _K_SS_MAXSIZE	128	/* Implementation specific max size */
-#define _K_SS_ALIGNSIZE	(__alignof__ (struct sockaddr *))
-				/* Implementation specific desired alignment */
 
 typedef unsigned short __kernel_sa_family_t;
 
-struct __kernel_sockaddr_storage {
-	__kernel_sa_family_t	ss_family;		/* address family */
-	/* Following field(s) are implementation specific */
-	char		__data[_K_SS_MAXSIZE - sizeof(unsigned short)];
-				/* space to achieve desired size, */
-				/* _SS_MAXSIZE value minus size of ss_family */
-} __attribute__ ((aligned(_K_SS_ALIGNSIZE)));	/* force desired alignment */
-
 #endif /* _LINUX_SOCKET_H */
diff --git a/include/netlink-private/netlink.h b/include/netlink-private/netlink.h
index d86a55d..ad24470 100644
--- a/include/netlink-private/netlink.h
+++ b/include/netlink-private/netlink.h
@@ -23,6 +23,7 @@
 #include <limits.h>
 #include <search.h>
 
+typedef uint32_t in_addr_t;
 #include <arpa/inet.h>
 #include <netdb.h>
 
diff --git a/src/nf-monitor.c b/src/nf-monitor.c
index 1f67289..2f84b00 100644
--- a/src/nf-monitor.c
+++ b/src/nf-monitor.c
@@ -5,6 +5,7 @@
  * Copyright (c) 2007 Secure Computing Corporation
  */
 
+#include <sys/select.h>
 #include <netlink/cli/utils.h>
 #include <netlink/netfilter/nfnl.h>
 
diff --git a/src/nf-queue.c b/src/nf-queue.c
index ec15f63..c37fc83 100644
--- a/src/nf-queue.c
+++ b/src/nf-queue.c
@@ -13,6 +13,7 @@
 #include <netlink/netfilter/queue.h>
 #include <netlink/netfilter/queue_msg.h>
 
+#include <sys/select.h>
 #include <linux/netlink.h>
 
 static struct nl_sock *nf_sock;
